scripts/setup/nxp/update_nxp_sdk.py

```diff
@@ -0,0 +1,98 @@
+    print("SDK will be cleaned all local modification(s) will be lost")
+    # Cleaning all local modifications
+    git_clean_command = "git reset --hard && git clean -xdf"
+    command = ['west', 'forall', '-c', git_clean_command, '-a']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+
+
+def init_platform_sdk_version(sdk_target_location, force):
+    print("Init SDK in: " + sdk_target_location)
+    if not force and os.path.exists(os.path.join(sdk_target_location, '.west')):
+        print("Error SDK is already initialized, use --force to force init")
+        print("WARNING -- All local SDK modification(s) will be lost")
+        sys.exit(1)
+    command = ['rm', '-rf', '.west']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+
+    command = ['west', 'init', '-l', 'manifest', '--mf', 'west.yml']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+    update_platform_sdk_version(sdk_target_location, force)
+
+
+def update_platform_sdk_version(sdk_target_location, force):
+    print("Update SDK in " + sdk_target_location)
+    if not os.path.exists(os.path.join(sdk_target_location, '.west')):
+        print("--update-only error SDK is not initialized")
+        sys.exit(1)
+    command = ['west', 'update', '--fetch', 'smart']
+    try:
+        subprocess.run(command, cwd=sdk_target_location, check=True)
+    except (RuntimeError, subprocess.CalledProcessError) as exception:
+        if force:
+            # In case of force update option and in case of update failure:
+            # 1. try to clean all local modications if any
+            # 2. Retry the west update command. It should be successfull now as all local modifications have been cleaned
+            clean_sdk_local_changes(sdk_target_location)
+            subprocess.run(command, cwd=sdk_target_location, check=True)
+        else:
+            print(exception)
+            print("Error SDK cannot be updated, local changes should be cleaned manually or use --force to force update")
+            print("WARNING -- With --force all local SDK modification(s) will be lost")
+
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Checkout or update relevant NXP SDK')
+    parser.add_argument(
+        "--update-only", help="Update NXP SDK to the correct version. Would fail if the SDK does not exist", action="store_true")
+    parser.add_argument('--platform', nargs='+', choices=all_platform_sdk_list, default=all_platform_sdk_list,
+                        help='Allows to select which SDK for a particular NXP platform to initialize')
+    parser.add_argument('--force', action='store_true',
+                        help='Force SDK initialization, hard clean will be done in case of failure - WARNING -- All local SDK modification(s) will be lost')
+
+    args = parser.parse_args()
+
+    for current_plat in args.platform:
+        plat_path = list(map(lambda res: res['plat_sdk_manifest_path'], filter(
```

Comment:
this should probably read like:

```py
plat_path = [p.manifest_path for p in ALL_PLATFORM_SDK if p.name == current_plat][0]
```

- use dataclasses
- using `[]` seems more readable than list/map/lambda